awx/main/utils/common.py

```diff
@@ -144,6 +146,18 @@ def underscore_to_camelcase(s):
     return ''.join(x.capitalize() or '_' for x in s.split('_'))
 
 
+def is_testing(argv=None):
```

Comment:
Could we use `@functools.cache` here?

Reply:
yes, that would be better. We use `cachetools` in other places, I don't have any examples where we use `functools.cache`, but I think it should be safe.

https://github.com/python/cpython/blob/f9433fff476aa13af9cb314fcc6962055faa4085/Lib/functools.py#L653

https://github.com/python/cpython/blob/f9433fff476aa13af9cb314fcc6962055faa4085/Lib/functools.py#L549

Since we won't ever _invalidate_ the cache, it could be thread-safe, or at least avoid the `del cache[key]` we hit before.

Reply:
I updated to use that, the import was reorganized a bit.